Provider panics when sentry is not available

It seems the provider is crashing whenever sentry can not respond in time or is not reachable.

This is the output of terraform:
```
╷
│ Error: Plugin did not respond
│ 
│   with data.sentry_organization.default,
│   on .terraform/data.tf line 1, in data "sentry_organization" "default":
│    1: data "sentry_organization" "default" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain
│ more details.
╵
Stack trace from the terraform-provider-sentry_v0.11.2 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xabd349]
goroutine 38 [running]:
github.com/jianyuan/go-sentry/v2/sentry.CheckResponse(0xbb3be0?)
	github.com/jianyuan/go-sentry/v2@v2.2.0/sentry/sentry.go:343 +0x29
github.com/jianyuan/terraform-provider-sentry/sentry.(*Config).Client.func1(0xd8cf10?, 0xc00004c130?, 0x15?, 0xd88c40?)
	github.com/jianyuan/terraform-provider-sentry/sentry/config.go:39 +0x30
github.com/hashicorp/go-retryablehttp.(*Client).Do(0xc00014a1c0, 0xc0002a2468)
	github.com/hashicorp/go-retryablehttp@v0.7.2/client.go:680 +0xd37
github.com/hashicorp/go-retryablehttp.(*RoundTripper).RoundTrip(0xc0000105b8, 0x0?)
	github.com/hashicorp/go-retryablehttp@v0.7.2/roundtripper.go:44 +0x7d
github.com/jianyuan/terraform-provider-sentry/sentry.(*semaphoreTransport).RoundTrip(0xc00052afc0, 0xc000248100)
	github.com/jianyuan/terraform-provider-sentry/sentry/config.go:83 +0x20f
net/http.send(0xc000248100, {0xd87940, 0xc00052afc0}, {0xc41440?, 0x417101?, 0x0?})
	net/http/client.go:251 +0x5f7
net/http.(*Client).send(0xc00052aff0, 0xc000248100, {0xc00053cf30?, 0x41a026?, 0x0?})
	net/http/client.go:175 +0x9b
net/http.(*Client).do(0xc00052aff0, 0xc000248100)
	net/http/client.go:715 +0x8fc
net/http.(*Client).Do(...)
	net/http/client.go:581
github.com/jianyuan/go-sentry/v2/sentry.(*Client).BareDo(0x0?, {0xd8cf48, 0xc000280a20}, 0xc0002d03c0?)
	github.com/jianyuan/go-sentry/v2@v2.2.0/sentry/sentry.go:242 +0x45
github.com/jianyuan/go-sentry/v2/sentry.(*Client).Do(0xc0002a41e0?, {0xd8cf48?, 0xc000280a20?}, 0xc000048900?, {0xb2c680?, 0xc0002d82c0})
	github.com/jianyuan/go-sentry/v2@v2.2.0/sentry/sentry.go:260 +0x79
github.com/jianyuan/go-sentry/v2/sentry.(*OrganizationsService).Get(0xc0002a4200, {0xd8cf48, 0xc000280a20}, {0xc0002a8194?, 0xc00053d2f8?})
	github.com/jianyuan/go-sentry/v2@v2.2.0/sentry/organizations.go:108 +0xe5
github.com/jianyuan/terraform-provider-sentry/sentry.dataSourceSentryOrganizationRead({0xd8cf48, 0xc000280a20}, 0x0?, {0xb9ea40?, 0xc0002a41e0})
	github.com/jianyuan/terraform-provider-sentry/sentry/data_source_sentry_organization.go:45 +0x1e5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc00049c000, {0xd8cf80, 0xc0002d2ab0}, 0xd?, {0xb9ea40, 0xc0002a41e0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:724 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc00049c000, {0xd8cf80, 0xc0002d2ab0}, 0xc00063e700, {0xb9ea40, 0xc0002a41e0})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:943 +0x145
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc000469278, {0xd8cf80?, 0xc0002d2990?}, 0xc0005c00e0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1179 +0x38f
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc000[303](https://gitlab.operations.justtrack-ops.io/backend/terminal-appsflyer/-/jobs/178125#L303)860, {0xd8cf80?, 0xc0002d21b0?}, 0xc0002d40a0)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:658 +0x403
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0xc2cf00?, 0xc000303860}, {0xd8cf80, 0xc0002d21b0}, 0xc00031a000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000480000, {0xd90980, 0xc0004e0000}, 0xc0002d6000, 0xc000492780, 0x1228130, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1[340](https://gitlab.operations.justtrack-ops.io/backend/terminal-appsflyer/-/jobs/178125#L340) +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc000480000, {0xd90980, 0xc0004e0000}, 0xc0002d6000, 0x0)
	google.golang.org/grpc@v1.50.1/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.50.1/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.50.1/server.go:963 +0x28a
Error: The terraform-provider-sentry_v0.11.2 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/internal/sentryclient/rate_limiter.go
+++ b/internal/sentryclient/rate_limiter.go
@@ -20,8 +20,10 @@ func NewRateLimiterRoundTripper(delegate http.RoundTripper) http.RoundTripper {
 	retryClient.ErrorHandler = retryablehttp.PassthroughErrorHandler
 	retryClient.Logger = nil // Disable DEBUG logs
 	retryClient.Backoff = func(min, max time.Duration, attemptNum int, resp *http.Response) time.Duration {
-		if rateLimitErr, ok := sentry.CheckResponse(resp).(*sentry.RateLimitError); ok {
-			return time.Until(rateLimitErr.Rate.Reset)
+		if resp != nil {
+			if rateLimitErr, ok := sentry.CheckResponse(resp).(*sentry.RateLimitError); ok {
+				return time.Until(rateLimitErr.Rate.Reset)
+			}
 		}
 		return retryablehttp.DefaultBackoff(min, max, attemptNum, resp)
 	}

